include/seastar/core/metrics.hh

```diff
@@ -404,8 +404,8 @@ public:
 instance_id_type shard();
 
 template<typename T, typename = std::enable_if_t<std::is_invocable_v<T>>>
-metric_function make_function(T val, data_type dt) {
-    return [dt, val] {
+metric_function make_function(T&& val, data_type dt) {
+    return [dt, val = std::move(val)] {
```

Comment:
If you make this a universal reference, you have to use std::forward here, otherwise it can move away from a non-temporary variable.

Reply:
It's better to keep the parameter by value. Since we need to make a copy anyway if we're passed a non-temporary. The move() is then safe.

Reply:
> If you make this a universal reference, you have to use std::forward here, otherwise it can move away from a non-temporary variable.

yeah. in theory, yes.

> Since we need to make a copy anyway if we're passed a non-temporary. The move() is then safe.

agreed. "if we're passed a non-temporary". changed accordingly.